frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/mobile_toolbar_v3/appflowy_mobile_toolbar_item.dart

```diff
@@ -59,12 +65,28 @@ class AppFlowyMobileToolbarIconItem extends StatefulWidget {
 class _AppFlowyMobileToolbarIconItemState
     extends State<AppFlowyMobileToolbarIconItem> {
   bool isSelected = false;
+  StreamSubscription? _subscription;
 
   @override
   void initState() {
     super.initState();
 
     isSelected = widget.isSelected?.call() ?? false;
+    if (widget.shouldListenToToggledStyle) {
+      widget.editorState.toggledStyleNotifier.addListener(_rebuild);
+      _subscription = widget.editorState.transactionStream.listen((_) {
+        _rebuild();
+      });
+    }
+  }
+
+  @override
+  void dispose() {
+    if (widget.shouldListenToToggledStyle) {
+      widget.editorState.toggledStyleNotifier.removeListener(_rebuild);
+      _subscription?.cancel();
+    }
```

Comment:
There's no way `widget.shouldListenToToggledStyle` can change? If it can, then we should consider adding some logic to `didUpdateWidget`.

Reply:
Yes. This value should not be changed after setting.